src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -73,7 +73,7 @@ internal sealed class LocalFunctionSymbol : MethodSymbol
             }
 
             _binder = binder;
-            _refKind = (syntax.ReturnType.Kind() == SyntaxKind.RefType) ? RefKind.Ref : RefKind.None;
+            _syntax.ReturnType.SkipRef(out _refKind);
```

Comment:
Also consider removing this field entirely and just computing it based on syntax every time. I bet the time/memory trade off here is not useful. Same with SourceMemberMethodSymbol.